app/routes/search/$text.tsx

```diff
@@ -31,22 +38,26 @@ export async function loader({ request, params }: LoaderArgs) {
     url.searchParams.get("caseSensitive") === "true"
   const pageNumber: string | undefined =
     url.searchParams.get("pageNumber") ?? undefined
+  const attribute: string =
+    url.searchParams.get("attribute") ?? SearchResultEnum.HEADWORD
 
-  const everything = await getSearchResultsByPage(
+  const searchResults = await getSearchResults(
     params.text,
     pageNumber,
-    caseSensitive
+    caseSensitive,
+    attribute
   )
 
-  if (!everything) {
+  if (!searchResults) {
```

Comment:
If there are no search results, we should probably not throw a 404; instead, show a page to the user indicating that there are no search results for their combination of filters.

Reply:
done

Reply:
I'm using an Error instead of a redirect, so the user still has all the search parameters the same in case they'd want to select a different attribute to search on (Headword instead of Meaning, for instance)